cpu/esp32/stdatomic.c

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2025 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp_common
+ * @{
+ *
+ * @file
+ * @brief       Missing atomic built-in functions
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @}
+ */
+
+#include <stdbool.h>
+
+#if (__GNUC__ == 14) && (__GNUC_MINOR__ == 2)
+
+#if defined(CPU_FAM_ESP32S2) || defined(CPU_FAM_ESP32C3)
```

Comment:
```suggestion
__attribute__((weak))
```

wouldn't this be more future-proof?

Reply:
I couldn't figure out exactly why `__atomic_test_and_set` is missing in GCC v14.2 for ESP32-S2 and ESP32-C3 while it exists for the other ESP32x variants :thinking: The problem was only mentioned in https://github.com/espressif/esp-idf/issues/15167.